### DISCUSSION TITLE
Display Validation Errors on TextBox

### DISCUSSION THREAD

**Discussion by djhohnstein** [AvaloniaCommunity/Material.Avalonia#401] [Q&A] [answered]
Hey everyone, new to Avalonia. Had a question about displaying validation errors and the MVVM toolkit.

I have the following ViewModel:

```c#
public class CustomViewModel : ViewModelBase {
    [ObservableProperty]
    [NotifyDataErrorInfo]
    [CustomValidation(typeof(SharedDefinitions), 
        nameof(SharedDefinitions.ValidateTextAlphaNumeric))]
    [Required]
    private string _name; 
...
}
```

Let's say I have the corresponding UserControl with the following defined:

```xml
<Grid RowDefinitions="*" ColumnDefinitions="Auto,*">
    <TextBlock
        VerticalAlignment="Center"
        FontSize="18"
        FontWeight="Bold"
        Margin="0 0 40 0"
        Grid.Row="0" Grid.Column="0" Text="Name:" />
    <TextBox 
        Grid.Row="0" Grid.Column="1" 
        assists:TextFieldAssist.Hints="{StaticResource AlphaNumericOnly}"
        Text="{Binding Name}" />
</Grid>
```

Upon the rendering of my view and entering of data into the `TextBox` element, I see that the outline of this box properly changes to red when a validation error occurs from the `ValidateTextAlphaNumeric` function (which returns a `ValidationResult` object), but the error message of that `ValidationResult` is **not** rendered on the corresponding TextBox hint. Is it possible to show the validation errors for that field on the text box?

**Comment by SKProCH** [comment, marked as answer]
Seems like this bug, since before it showns below the textbox. I'll take a look. You can track progress at https://github.com/AvaloniaCommunity/Material.Avalonia/issues/402

**Reply by djhohnstein** [reply]
Awesome thank you!
